src/execution_plan/execution_plan_modify.c

```diff
@@ -128,15 +128,15 @@ void ExecutionPlan_RemoveOp(ExecutionPlan *plan, OpBase *op) {
 		// Remove new root's parent pointer.
 		plan->root->parent = NULL;
 	} else {
-		if(op->childCount == 1) {
+		OpBase *parent = op->parent;
+		if(op->childCount > 0) {
+			// In place replacement of the op bounded branch instead of op.
```

Comment:
Don't use the term bound branch, as it is specific for only a few operations.

Reply:
changed